pkg/apis/core/validation/shoot.go

```diff
@@ -1343,6 +1343,9 @@ func ValidateKubeletConfig(kubeletConfig core.KubeletConfig, version string, doc
 			allErrs = append(allErrs, field.Forbidden(fldPath.Child("seccompDefault"), "seccomp defaulting is not available when kubelet's 'SeccompDefault' feature gate is disabled"))
 		}
 	}
+	if v := kubeletConfig.ContainerLogMaxFiles; v != nil && (*v < 2) {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("containerLogMaxFiles"), *v, "value must be >= 2."))
+	}
```

Comment:
I am missing validation of the new fields in `.spec.kubernetes.kubelet`. Concretely, we should only allow setting the new fields when ALL worker pools are using `containerd`. When there is at least one pool with `docker` we should disallow setting them there centrally and rather recommend to use the pool-specific kubelet settings.